creator-node/src/premiumContent/helpers.ts

```diff
@@ -0,0 +1,59 @@
+import type Logger from 'bunyan'
+import { PremiumContentSignatureData, PremiumContentType } from './types'
+
+type PremiumContentMatchArgs = {
+  signedDataFromDiscoveryNode: PremiumContentSignatureData
+  userWallet: string
+  premiumContentId: number
+  premiumContentType: PremiumContentType
+  logger?: any
+}
+
+const PREMIUM_CONTENT_SIGNATURE_MAX_TTL_MS = 6 * 60 * 60 * 1000 // 6 hours
+
+export const isPremiumContentMatch = ({
+  signedDataFromDiscoveryNode,
+  userWallet,
+  premiumContentId,
+  premiumContentType,
+  logger = console
+}: PremiumContentMatchArgs) => {
+  const {
+    premiumContentId: signedPremiumContentId,
+    premiumContentType: signedPremiumContentType,
+    userWallet: signedUserWallet,
+    timestamp: signedTimestamp
+  } = signedDataFromDiscoveryNode
+
+  const isSignatureTooOld =
+    Date.now() - signedTimestamp >= PREMIUM_CONTENT_SIGNATURE_MAX_TTL_MS
```

Comment:
let's use this fn to check expiration https://github.com/AudiusProject/audius-protocol/pull/3598/files#diff-d8f9eba7f913c83915368ff67a4aaaf31db348963f3c1673c81690a85b868ab9R76-R84. I added a param to pass in the ttl / pr should be merged soon